## Import necessary libraries

In [1]:
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns

## Import dataset 

In [2]:
df = pd.read_csv('1.csv')

## Exploratory Data Analysis and Data Cleaning

In [3]:
df.head()

id    surname  firstname middlename  gender  gender_male  gender_female  \
0   1     Solade      Grace     Solape  Female            0              1   
1   2     Eneche       Kure    Balogun    Male            1              0   
2   3     Sanusi     Adaugo   Katerine  Female            0              1   
3   4     Sowore  Mooslemat    Ifedayo  Female            0              1   
4   5  Abdusalam      Yusuf     Okafor    Male            1              0   

    state settlement  rural_settlement  ...  NmA NmC  NmW  health_status  \
0  Rivers      Rural                 1  ...    0   0    0          alive   
1  Ebonyi      Rural                 1  ...    0   0    1          Alive   
2    Ogun      Urban                 0  ...    0   0    1           Dead   
3    Ondo      Rural                 1  ...    0   0    1          Alive   
4     Oyo      Urban                 0  ...    0   0    1          Alive   

  alive dead  report_outcome  unconfirmed confirmed null_serotype  
0     1    0       confirmed            0         1             1  
1     1    0       Confirmed            1         0             1  
2     0    1   Not Confirmed            1         0             1  
3     1    0   Not Confirmed            1         0             1  
4     1    0       Confirmed            1         0             1  

[5 rows x 41 columns]

In [4]:
df.columns #

Index(['id', 'surname', 'firstname', 'middlename', 'gender', 'gender_male',
       'gender_female', 'state', 'settlement', 'rural_settlement',
       'urban_settlement', 'report_date', 'report_year', 'age', 'age_str',
       'date_of_birth', 'child_group', 'adult_group', 'Age Classification',
       'disease', 'cholera', 'diarrhoea', 'measles',
       'viral_haemmorrhaphic_fever', 'meningitis', 'ebola', 'marburg_virus',
       'yellow_fever', 'rubella_mars', 'malaria', 'serotype', 'NmA', 'NmC',
       'NmW', 'health_status', 'alive', 'dead', 'report_outcome',
       'unconfirmed', 'confirmed', 'null_serotype'],
      dtype='object')

In [5]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 284484 entries, 0 to 284483
Data columns (total 41 columns):
 #   Column                      Non-Null Count   Dtype 
---  ------                      --------------   ----- 
 0   id                          284484 non-null  int64 
 1   surname                     284484 non-null  object
 2   firstname                   284484 non-null  object
 3   middlename                  284484 non-null  object
 4   gender                      284484 non-null  object
 5   gender_male                 284484 non-null  int64 
 6   gender_female               284484 non-null  int64 
 7   state                       284484 non-null  object
 8   settlement                  284484 non-null  object
 9   rural_settlement            284484 non-null  int64 
 10  urban_settlement            284484 non-null  int64 
 11  report_date                 284484 non-null  object
 12  report_year                 284484 non-null  int64 
 13  age                         2

### change the name of Age Classification to follow the naming convention as seen in the dataset

In [6]:
df.rename(columns = {'Age Classification':'age_classification'}, inplace=True)

### confirm the changes made.

In [7]:
df.columns

Index(['id', 'surname', 'firstname', 'middlename', 'gender', 'gender_male',
       'gender_female', 'state', 'settlement', 'rural_settlement',
       'urban_settlement', 'report_date', 'report_year', 'age', 'age_str',
       'date_of_birth', 'child_group', 'adult_group', 'age_classification',
       'disease', 'cholera', 'diarrhoea', 'measles',
       'viral_haemmorrhaphic_fever', 'meningitis', 'ebola', 'marburg_virus',
       'yellow_fever', 'rubella_mars', 'malaria', 'serotype', 'NmA', 'NmC',
       'NmW', 'health_status', 'alive', 'dead', 'report_outcome',
       'unconfirmed', 'confirmed', 'null_serotype'],
      dtype='object')

### checking for incorrect values in the report date field

In [8]:
df[df['report_date'] == '0000-00-00']

id   surname  firstname middlename  gender  gender_male  \
23          24   Bayeero     Areoye    Adesina    Male            1   
209250  213189   Omotoni  Benedicta  Adetorera  Female            0   
209256  213195  Omatsope    Danjuma      Okeke    Male            1   
209307  213246  Omotsope   Benedict   Akindele    Male            1   
209388  213327     Asibi    Paulina    Kehinde  Female            0   
...        ...       ...        ...        ...     ...          ...   
284122  288061    Ladoke     Alimat     Osahon  Female            0   
284138  288077  Okerewke    Danjuma     Ojukwu    Male            1   
284150  288089     Jimoh      Simbi      Grace  Female            0   
284330  288269    Sunday     Adaugo    Kehinde  Female            0   
284459  288398  Folawiyo     Adijah   Folasade  Female            0   

        gender_female     state settlement  rural_settlement  ...  NmA NmC  \
23                  0      Ogun      Rural                 1  ...    0   0   
209250              1     Kebbi      Urban                 0  ...    0   0   
209256              0    Rivers      Rural                 1  ...    0   0   
209307              0   Zamfara      Urban                 0  ...    0   0   
209388              1     Lagos      Urban                 0  ...    0   0   
...               ...       ...        ...               ...  ...  ...  ..   
284122              1  Nasarawa      Urban                 0  ...    0   1   
284138              0    Taraba      Rural                 1  ...    0   0   
284150              1     Kwara      Urban                 0  ...    0   0   
284330              1   Katsina      Urban                 0  ...    0   0   
284459              1     Kwara      Rural                 1  ...    0   0   

        NmW  health_status alive dead  report_outcome  unconfirmed confirmed  \
23        1           Dead     0    1       Confirmed            0         1   
209250    1          Alive     1    0   Not Confirmed            1         0   
209256    1           Dead     0    1       Confirmed            1         0   
209307    1          Alive     1    0       Confirmed            1         0   
209388    1           Dead     0    1       Confirmed            1         0   
...     ...            ...   ...  ...             ...          ...       ...   
284122    0          Alive     1    0       Confirmed            1         0   
284138    1          Alive     1    0   Not Confirmed            1         0   
284150    1          Alive     1    0   Not Confirmed            1         0   
284330    1           Dead     0    1       Confirmed            0         1   
284459    1           Dead     0    1       Confirmed            0         1   

       null_serotype  
23                 1  
209250             1  
209256             1  
209307             1  
209388             1  
...              ...  
284122             0  
284138             1  
284150             1  
284330             1  
284459             1  

[1372 rows x 41 columns]

#### drop invalid data points, which totalled 1372 values, this number does not affect the general outlook of the dataset

In [9]:
df = df[df['report_date'] != '0000-00-00'] 

#### confirm the changes, we made to the dataset

In [10]:
df[df['report_date'] == '0000-00-00']

Empty DataFrame
Columns: [id, surname, firstname, middlename, gender, gender_male, gender_female, state, settlement, rural_settlement, urban_settlement, report_date, report_year, age, age_str, date_of_birth, child_group, adult_group, age_classification, disease, cholera, diarrhoea, measles, viral_haemmorrhaphic_fever, meningitis, ebola, marburg_virus, yellow_fever, rubella_mars, malaria, serotype, NmA, NmC, NmW, health_status, alive, dead, report_outcome, unconfirmed, confirmed, null_serotype]
Index: []

[0 rows x 41 columns]

#### as earlier stated, the values dropped does not affect the general outlook on the dataset 

In [11]:
df.count()

id                            283112
surname                       283112
firstname                     283112
middlename                    283112
gender                        283112
gender_male                   283112
gender_female                 283112
state                         283112
settlement                    283112
rural_settlement              283112
urban_settlement              283112
report_date                   283112
report_year                   283112
age                           283112
age_str                       283112
date_of_birth                 283112
child_group                   283112
adult_group                   283112
age_classification            283112
disease                       283112
cholera                       283112
diarrhoea                     283112
measles                       283112
viral_haemmorrhaphic_fever    283112
meningitis                    283112
ebola                         283112
marburg_virus                 283112
y

### check for invalid data in the date of birth column

In [12]:
df[df['date_of_birth']=='0000-00-00']['date_of_birth']

50        0000-00-00
115       0000-00-00
140       0000-00-00
178       0000-00-00
376       0000-00-00
             ...    
284259    0000-00-00
284272    0000-00-00
284282    0000-00-00
284298    0000-00-00
284323    0000-00-00
Name: date_of_birth, Length: 5032, dtype: object

### we now drop these data points, which totalled 5032, this does not have an effect on the dataset

In [13]:
df = df[df['date_of_birth'] != '0000-00-00'] 

#### confirm the changes we made to the dataset

In [14]:
df[df['date_of_birth']=='0000-00-00']

Empty DataFrame
Columns: [id, surname, firstname, middlename, gender, gender_male, gender_female, state, settlement, rural_settlement, urban_settlement, report_date, report_year, age, age_str, date_of_birth, child_group, adult_group, age_classification, disease, cholera, diarrhoea, measles, viral_haemmorrhaphic_fever, meningitis, ebola, marburg_virus, yellow_fever, rubella_mars, malaria, serotype, NmA, NmC, NmW, health_status, alive, dead, report_outcome, unconfirmed, confirmed, null_serotype]
Index: []

[0 rows x 41 columns]

### to count the number of entries left in the dataset

In [15]:
df.count()

id                            278080
surname                       278080
firstname                     278080
middlename                    278080
gender                        278080
gender_male                   278080
gender_female                 278080
state                         278080
settlement                    278080
rural_settlement              278080
urban_settlement              278080
report_date                   278080
report_year                   278080
age                           278080
age_str                       278080
date_of_birth                 278080
child_group                   278080
adult_group                   278080
age_classification            278080
disease                       278080
cholera                       278080
diarrhoea                     278080
measles                       278080
viral_haemmorrhaphic_fever    278080
meningitis                    278080
ebola                         278080
marburg_virus                 278080
y

### Now, we drop fields/ columns, we consider irrelevant to this analysis, or that would lead to redundance

In [16]:
df.drop(['gender_male','gender_female','rural_settlement','urban_settlement','child_group', 'adult_group','age_str','cholera', 'diarrhoea', 'measles','viral_haemmorrhaphic_fever', 'meningitis', 'ebola', 'marburg_virus', 'yellow_fever', 'rubella_mars', 'malaria', 'NmA', 'NmC', 'NmW', 'alive', 'dead', 'unconfirmed', 'confirmed', 'null_serotype'], axis=1, inplace=True)

### changing the index of the dataset

In [17]:
df.reset_index(inplace=True)

In [18]:
df.drop('id', axis=1, inplace=True)

In [19]:
df

index    surname  firstname middlename  gender        state  \
0            0     Solade      Grace     Solape  Female       Rivers   
1            1     Eneche       Kure    Balogun    Male       Ebonyi   
2            2     Sanusi     Adaugo   Katerine  Female         Ogun   
3            3     Sowore  Mooslemat    Ifedayo  Female         Ondo   
4            4  Abdusalam      Yusuf     Okafor    Male          Oyo   
...        ...        ...        ...        ...     ...          ...   
278075  284479    Mukthar     Alimat    Ajibewa  Female  Cross River   
278076  284480    Akenzua    Deborah   Rofeeqat  Female        Kwara   
278077  284481        Isa    Deborah    Tofunmi  Female      Katsina   
278078  284482        Uba     Ismail    Akpabio    Male       Jigawa   
278079  284483       Alao    Paulina       Sade  Female         Kogi   

       settlement report_date  report_year  age date_of_birth  \
0           Rural  15/05/2018         2018   32    17/01/1986   
1           Rural  28/02/2017         2017   32    01/09/1985   
2           Urban  02/03/2012         2012   24    01/05/1988   
3           Rural  21/05/2010         2010   63    24/05/1947   
4           Urban  28/08/2017         2017    9    07/01/2008   
...           ...         ...          ...  ...           ...   
278075      Rural  24/08/2012         2012   67    10/07/1945   
278076      Urban  19/01/2009         2009   64    27/01/1945   
278077      Rural  10/06/2010         2010   33    03/10/1977   
278078      Rural  11/01/2016         2016   18    07/05/1998   
278079      Urban  18/08/2013         2013    7    07/01/2013   

       age_classification        disease       serotype health_status  \
0                   Adult        Cholera  null serotype         alive   
1                   Adult  Marburg Virus           Null         Alive   
2                   Adult  Marburg Virus           Null          Dead   
3                   Adult        Measles           Null         Alive   
4                   Child   Rubella Mars           Null         Alive   
...                   ...            ...            ...           ...   
278075              Adult          Ebola           Null          Dead   
278076              Adult          Ebola           Null          Dead   
278077              Adult        Malaria           Null         Alive   
278078              Adult        Malaria           Null          Dead   
278079              Child   Yellow Fever           Null         Alive   

       report_outcome  
0           confirmed  
1           Confirmed  
2       Not Confirmed  
3       Not Confirmed  
4           Confirmed  
...               ...  
278075      Confirmed  
278076      Confirmed  
278077      Confirmed  
278078  Not Confirmed  
278079      Confirmed  

[278080 rows x 16 columns]

### We check for unique values in the serotype field/ column

In [20]:
df['serotype'].unique()

array(['null serotype', 'Null', 'Neisseria meningitidis group A (NmA)',
       'Neisseria meningitidis group W (NmW)',
       'Neisseria meningitidis group C (NmC)'], dtype=object)

#### We replace null serotype to Null, to give uniformity.

In [21]:
df['serotype'] = df['serotype'].replace(['null serotype'],'Null')

#### Check to confirm the changes we implemented.

In [22]:
df['serotype'].unique()

array(['Null', 'Neisseria meningitidis group A (NmA)',
       'Neisseria meningitidis group W (NmW)',
       'Neisseria meningitidis group C (NmC)'], dtype=object)

### We check for unique values in report outcome

In [23]:
df['report_outcome'].unique()

array(['confirmed', 'Confirmed', 'Not Confirmed'], dtype=object)

#### Now, we replace confirmed to Confirmed, for uniformity

In [24]:
df['report_outcome'] = df['report_outcome'].replace(['confirmed'],'Confirmed')

#### Check to confirm the changes we made

In [25]:
df['report_outcome'].unique()

array(['Confirmed', 'Not Confirmed'], dtype=object)

### We check the unique values in health status

In [26]:
df['health_status'].unique()

array(['alive', 'Alive', 'Dead'], dtype=object)

#### We replace alive to Alive, for uniformity

In [27]:
df['health_status'] = df['health_status'].replace(['alive'],'Alive')

#### Check to confirm changes

In [28]:
df['health_status'].unique()

array(['Alive', 'Dead'], dtype=object)

### Convert to necessary data types

In [29]:
df['report_date'] = pd.to_datetime(df['report_date'])

In [30]:
df['date_of_birth'] = pd.to_datetime(df['date_of_birth'])

#### Check to confirm changes made

In [31]:
df.dtypes

index                          int64
surname                       object
firstname                     object
middlename                    object
gender                        object
state                         object
settlement                    object
report_date           datetime64[ns]
report_year                    int64
age                            int64
date_of_birth         datetime64[ns]
age_classification            object
disease                       object
serotype                      object
health_status                 object
report_outcome                object
dtype: object

### Checking to see the number of confirmed and not comfirmed reports

In [32]:
df[df['report_outcome'] == 'Confirmed']

index    surname firstname middlename  gender        state  \
0            0     Solade     Grace     Solape  Female       Rivers   
1            1     Eneche      Kure    Balogun    Male       Ebonyi   
4            4  Abdusalam     Yusuf     Okafor    Male          Oyo   
5            5     Yakubu     Janet     Chioma  Female       Kaduna   
6            6      Razak    Adaugo     Adaobi  Female       Taraba   
...        ...        ...       ...        ...     ...          ...   
278074  284478    Adedayo     Judah     Adeoye    Male         Yobe   
278075  284479    Mukthar    Alimat    Ajibewa  Female  Cross River   
278076  284480    Akenzua   Deborah   Rofeeqat  Female        Kwara   
278077  284481        Isa   Deborah    Tofunmi  Female      Katsina   
278079  284483       Alao   Paulina       Sade  Female         Kogi   

       settlement report_date  report_year  age date_of_birth  \
0           Rural  2018-05-15         2018   32    1986-01-17   
1           Rural  2017-02-28         2017   32    1985-01-09   
4           Urban  2017-08-28         2017    9    2008-07-01   
5           Rural  2012-06-27         2012   44    1968-06-06   
6           Rural  2010-01-04         2010   61    1949-07-19   
...           ...         ...          ...  ...           ...   
278074      Urban  2010-04-11         2010   22    2014-11-26   
278075      Rural  2012-08-24         2012   67    1945-10-07   
278076      Urban  2009-01-19         2009   64    1945-01-27   
278077      Rural  2010-10-06         2010   33    1977-03-10   
278079      Urban  2013-08-18         2013    7    2013-07-01   

       age_classification        disease serotype health_status report_outcome  
0                   Adult        Cholera     Null         Alive      Confirmed  
1                   Adult  Marburg Virus     Null         Alive      Confirmed  
4                   Child   Rubella Mars     Null         Alive      Confirmed  
5                   Adult          Ebola     Null         Alive      Confirmed  
6                   Adult   Yellow Fever     Null         Alive      Confirmed  
...                   ...            ...      ...           ...            ...  
278074              Adult  Marburg Virus     Null         Alive      Confirmed  
278075              Adult          Ebola     Null          Dead      Confirmed  
278076              Adult          Ebola     Null          Dead      Confirmed  
278077              Adult        Malaria     Null         Alive      Confirmed  
278079              Child   Yellow Fever     Null         Alive      Confirmed  

[138524 rows x 16 columns]

In [33]:
df[df['report_outcome'] == 'Not Confirmed']

index    surname  firstname middlename  gender        state  \
2            2     Sanusi     Adaugo   Katerine  Female         Ogun   
3            3     Sowore  Mooslemat    Ifedayo  Female         Ondo   
7            7    Annakyi  Danmbazzu     Osagie    Male      Katsina   
10          10  Muhammadu     Adijah   Cathrine  Female        Niger   
11          11   Okerewke       Kure     Osahon    Male  Cross River   
...        ...        ...        ...        ...     ...          ...   
278068  284472    Adeyemi     Mariam      Funke  Female         Osun   
278070  284474    Fidelis      Yusuf        Oni    Male          Oyo   
278072  284476   Akintola      Debby   Patricia  Female       Taraba   
278073  284477      Gombi     Zainab   Akinremi  Female         Ondo   
278078  284482        Uba     Ismail    Akpabio    Male       Jigawa   

       settlement report_date  report_year  age date_of_birth  \
2           Urban  2012-02-03         2012   24    1988-01-05   
3           Rural  2010-05-21         2010   63    1947-05-24   
7           Rural  2015-04-10         2015    2    2014-12-18   
10          Urban  2015-12-05         2015   17    1998-07-06   
11          Rural  2012-09-28         2012   50    1962-04-02   
...           ...         ...          ...  ...           ...   
278068      Urban  2009-03-10         2009   43    1966-07-03   
278070      Urban  2009-12-11         2009    1    2013-06-10   
278072      Rural  2009-02-10         2009   69    1940-03-17   
278073      Urban  2015-05-14         2015   42    1973-04-13   
278078      Rural  2016-11-01         2016   18    1998-07-05   

       age_classification                     disease  \
2                   Adult               Marburg Virus   
3                   Adult                     Measles   
7                   Child                   Diarrhoea   
10                  Child                     Cholera   
11                  Adult                   Diarrhoea   
...                   ...                         ...   
278068              Adult                Yellow Fever   
278070              Child                  Meningitis   
278072              Adult                   Diarrhoea   
278073              Adult  Viral Haemmorrhaphic Fever   
278078              Adult                     Malaria   

                                    serotype health_status report_outcome  
2                                       Null          Dead  Not Confirmed  
3                                       Null         Alive  Not Confirmed  
7                                       Null         Alive  Not Confirmed  
10                                      Null         Alive  Not Confirmed  
11                                      Null         Alive  Not Confirmed  
...                                      ...           ...            ...  
278068                                  Null          Dead  Not Confirmed  
278070  Neisseria meningitidis group W (NmW)          Dead  Not Confirmed  
278072                                  Null          Dead  Not Confirmed  
278073                                  Null          Dead  Not Confirmed  
278078                                  Null          Dead  Not Confirmed  

[139556 rows x 16 columns]

### Let's get a percentage to confirmed and unconfirmed reports

In [34]:
round( (df[df['report_outcome'] == 'Confirmed'].count() / df.count()) * 100 ,2)

index                 49.81
surname               49.81
firstname             49.81
middlename            49.81
gender                49.81
state                 49.81
settlement            49.81
report_date           49.81
report_year           49.81
age                   49.81
date_of_birth         49.81
age_classification    49.81
disease               49.81
serotype              49.81
health_status         49.81
report_outcome        49.81
dtype: float64

In [35]:
round( (df[df['report_outcome'] == 'Not Confirmed'].count() / df.count()) * 100 ,2)

index                 50.19
surname               50.19
firstname             50.19
middlename            50.19
gender                50.19
state                 50.19
settlement            50.19
report_date           50.19
report_year           50.19
age                   50.19
date_of_birth         50.19
age_classification    50.19
disease               50.19
serotype              50.19
health_status         50.19
report_outcome        50.19
dtype: float64

#### As we can see, there are more uncomfirmed reports than confirmed ones, as given by 50.19% to 49.81%, a slight but significant difference

### Let us check the start date and end date of this dataset using the report date field

In [36]:
df['report_date'].min()

Timestamp('2009-01-01 00:00:00')

In [37]:
df['report_date'].max()

Timestamp('2018-12-31 00:00:00')

#### As we can see, there data was collected for 9 years, from 01-Jan-2009 to 31-Dec-2018, this also serves as the oldest and latest records in the dataset

### Let us check what year had the most number of cases

In [38]:
grouped_by_year = df.groupby('report_year').size().sort_values(ascending=False)
grouped_by_year

report_year
2013    28089
2016    28025
2010    27919
2017    27902
2018    27831
2012    27809
2014    27742
2009    27665
2015    27581
2011    27517
dtype: int64

#### As we can see, the year 2013, had the most number of cases.

### Now, let's see the most prevalent diesease in the country

In [39]:
grouped_by_disease = df.groupby('disease').size().sort_values(ascending=False)
grouped_by_disease

disease
Cholera                       27935
Rubella Mars                  27875
Malaria                       27873
Diarrhoea                     27872
Measles                       27831
Marburg Virus                 27826
Yellow Fever                  27806
Viral Haemmorrhaphic Fever    27777
Meningitis                    27730
Ebola                         27555
dtype: int64

#### Cholera is the most prevalent disease followed by Rubella Mars

### Let's dive deep into the various diseases that were rambant in 2013, as it had the highest number of records from this dataset, we'd also check report was confirmed or not

In [40]:
df[df['report_year']== 2013].groupby(['disease','report_outcome']).size().sort_values(ascending=False)

disease                     report_outcome
Cholera                     Not Confirmed     1445
Malaria                     Confirmed         1442
Viral Haemmorrhaphic Fever  Not Confirmed     1433
Diarrhoea                   Not Confirmed     1432
Ebola                       Confirmed         1432
Measles                     Not Confirmed     1430
Cholera                     Confirmed         1425
Rubella Mars                Not Confirmed     1425
Meningitis                  Not Confirmed     1417
                            Confirmed         1413
Yellow Fever                Confirmed         1404
Marburg Virus               Not Confirmed     1403
Malaria                     Not Confirmed     1392
Diarrhoea                   Confirmed         1388
Measles                     Confirmed         1384
Ebola                       Not Confirmed     1377
Marburg Virus               Confirmed         1375
Viral Haemmorrhaphic Fever  Confirmed         1374
Rubella Mars                Confirmed  

#### From the above outcome, we can see that uncomfirmed cholera cases, tops the chart of diseases reported in 2013

### Let's get what disease caused the most casualty

In [41]:
df[df['health_status'] == 'Dead'].groupby(['disease','report_year']).size().sort_values(ascending=False)

disease                     report_year
Ebola                       2014           1485
Malaria                     2017           1467
Diarrhoea                   2013           1461
Cholera                     2012           1458
                            2016           1442
                                           ... 
                            2010           1336
Ebola                       2009           1329
Cholera                     2014           1323
Viral Haemmorrhaphic Fever  2018           1315
Malaria                     2012           1312
Length: 100, dtype: int64

#### Ebola in the year 2014 had the most casulties, as a result of the Ebola outbreak, with Malaria causing the least deaths in 2012

### Mortality rate per diseases

In [42]:
round( ((df[df['health_status'] == 'Dead'].groupby('disease').size()/df.groupby('disease').size()) *100) ,2).sort_values(ascending=False)

disease
Ebola                         50.32
Diarrhoea                     50.29
Cholera                       50.15
Viral Haemmorrhaphic Fever    50.08
Marburg Virus                 50.01
Meningitis                    50.00
Rubella Mars                  49.97
Yellow Fever                  49.97
Measles                       49.90
Malaria                       49.63
dtype: float64

In [43]:
round(((df[df['health_status'] == 'Dead']['disease'].count() / df['disease'].count()) *100),2)

50.03

#### As we can see, Ebola has the highest mortality rate of 50.32%, with Malaria having the lowest mortality rate of 49.63%, but in general all the diseases have a quite high mortality rate, with an average mortality rate of 50.03%.

### Mortality Rate per year per state

In [44]:
df[df['health_status'] == 'Dead'].groupby(['report_year','state']).size().sort_values(ascending=False)

report_year  state                    
2010         Federal Capital Territory    434
2009         Taraba                       433
2013         Bayelsa                      429
             Ekiti                        424
2009         Ondo                         424
                                         ... 
2016         Akwa Ibom                    334
2009         Ekiti                        334
2014         Plateau                      333
2018         Edo                          330
2015         Ekiti                        329
Length: 370, dtype: int64

#### From the above output, it is now known that the FCT in the year 2010, recorded the highest number of deathe, with Ekiti in 2015, recording the lowest number of deaths

### Number of cases categorized by state

In [45]:
df.groupby(['state','disease']).size().sort_values(ascending=False)

state    disease      
Abia     Marburg Virus    831
Benue    Meningitis       829
Kwara    Measles          827
Kano     Malaria          818
Borno    Yellow Fever     815
                         ... 
Bayelsa  Yellow Fever     693
Ekiti    Cholera          687
Ogun     Rubella Mars     684
Kogi     Measles          683
Kwara    Meningitis       680
Length: 370, dtype: int64

In [46]:
df.groupby(['state','report_year']).size().sort_values(ascending=False)

state     report_year
Niger     2013           837
Rivers    2012           835
Niger     2017           824
Sokoto    2009           811
Ondo      2018           807
                        ... 
Plateau   2011           697
Nasarawa  2015           693
Plateau   2014           683
Rivers    2013           678
Sokoto    2015           672
Length: 370, dtype: int64

In [47]:
df.groupby('state').size().sort_values(ascending=False)

state
Niger                        7676
Ondo                         7674
Kano                         7663
Bayelsa                      7612
Kebbi                        7601
Imo                          7601
Benue                        7593
Ebonyi                       7590
Yobe                         7584
Kwara                        7555
Nasarawa                     7553
Osun                         7550
Kaduna                       7539
Jigawa                       7537
Anambra                      7536
Enugu                        7534
Oyo                          7529
Abia                         7522
Sokoto                       7512
Delta                        7506
Adamawa                      7502
Federal Capital Territory    7502
Gombe                        7497
Kogi                         7495
Rivers                       7486
Lagos                        7482
Bauchi                       7482
Zamfara                      7463
Ogun                         7452
Edo     

#### From the above outputs, we can see that Niger State had the highest number of records in general, also in 2013, the same state recorded the highest number of records, with Cross River recording the lowest number of overall record, with Sokoto State recording the least record in a calendar year, 2015, Abia State recorded the most cases from a disease: Marburg Virus, and Kwara State recording the least records for a particular disease: Meningitis. 